### PR TITLE
feature: add memberships endpoint to the API transport

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request adds a new `Memberships` endpoint with API transport support.

Requires https://github.com/goabstract/projects/pull/1977
References #108

### CODE REVIEW

File: docs/abstract-api.md [hunk 1]
```diff
@@ -1191,3 +1234,13 @@ Reference for the parameters required to load resources with Abstract SDK.
   projectId: string
 }
 ```
+
+### MembershipDescriptor
+
+```js
+{
+  organizationId?: string,
+  projectId?: string,
```
Comment (amccloud) [inline docs/abstract-api.md]: I think we can drop these optional props and rename these to `UserDescriptor`. Then the endpoints can be typed as: 

```js
list(objectDescriptor: OrganizationDescriptor | ProjectDescriptor | UserDescriptor) {}

// and

info(userDescriptor: UserDescriptor) {}
```

Comment (bitpshr) [reply docs/abstract-api.md]: Much cleaner, great suggestion. Updated.

File: docs/abstract-api.md [hunk 2]
```diff
@@ -1191,3 +1234,21 @@ Reference for the parameters required to load resources with Abstract SDK.
...
+  userId: string
+}
+```
+
+### ProjectMembershipDescriptor
+
+```js
+{
+  organizationId?: string,
```
Comment (amccloud) [inline docs/abstract-api.md]: Does `organizationId` and `projectId` (above) still need to be nullable?

File: src/types.js [hunk 3]
```diff
@@ -328,6 +328,27 @@ export type User = {
...
+
+export type ProjectMembershipDescriptor = {|
+  ...$Exact<MembershipDescriptorBase>,
+  projectId: string
+|};
+
+export type OrganizationMembershipDescriptor = {|
+  ...$Exact<MembershipDescriptorBase>,
+  organizationId: string
+|};
```
Comment (amccloud) [inline src/types.js]: Do we still need `MembershipDescriptorBase`? Seem like `userId` can just be hardcoded in each type

File: src/types.js [hunk 4]
```diff
@@ -328,6 +328,27 @@ export type User = {
...
+
+export type Membership = {
+  createdAt: string,
+  organizationId?: string,
+  projectId?: string,
```
Comment (amccloud) [inline src/types.js]: When are these undefined? Seems like at least `organizaionId` should always be set

Comment (bitpshr) [reply src/types.js]: Good catch. `projectId` can be undefined for an organization-only membership, but you're totally right about `organizationId` always being populated. Updated.

**Pull request merged**